### DISCUSSION TITLE
Integracja Domoticza z miarkownikiem

### DISCUSSION THREAD

**Discussion by krzychoooo** [krzychoooo/Miarkownik#12] [General]
Tutaj piszemy o integracji Domoticza z miarkownikiem.

**Comment by krzychoooo** [comment]
Generalnie w Domoticz tworzymy sprzęt typu Dummy a w nim czujkę typu temperatura. Przechodzimy do urządzeń i odczytujemy IDx czujki.
W miarkowniku -> Devices -> Environment - DS18b20 -> Edit 
![elektroniczny miarkownik ciągu WiFi 2](https://user-images.githubusercontent.com/7173366/103033233-4fa11380-4562-11eb-9812-00a170474ad1.PNG)
Zaznaczamy "Send to Controller " i wybieramy IDX zapamiętany z Domoticza. 
Cieszymy się w Domoticz z pokazywanej temperatury pieca.

**Reply by piotrux** [reply]
Wszytko działa, dziękuję. Teraz pytanie level pro, czy jakoś da się odpalić np taki wirtualny termostat z wykorzystaniem np netatmo. Netatmo mam i działa pięknie z domoticzem. Są takie pluginy tylko wykorzystują on/off i czujki wewnętrzne np https://www.domoticz.com/wiki/Plugins/Smart_Virtual_Thermostat.html
![Screenshot_20201223_231824](https://user-images.githubusercontent.com/76114450/103040979-55ecbb00-4575-11eb-8e5f-efc8608f2c89.jpg)

**Comment by piotrux** [comment]
Ok odpaliłem domoticza ... i jestem trochę zielony. W miarkowniku podałem IP i port domoticza, dalej utknąłem

**Reply by krzychoooo** [reply]
W kontrollers -> Domoticz HTTP -> edit
na dole musi być **Enabled**
![elektroniczny miarkownik ciągu WiFi](https://user-images.githubusercontent.com/7173366/103040737-add6f200-4574-11eb-9421-98224a8f46f7.PNG)

**Comment by piotrux** [comment]
Jeszcze jest Home Assistant, może w wątku także dodać to oprogramowanie?

**Reply by krzychoooo** [reply]
Nie bardzo rozumiem?

**Reply by krzychoooo** [reply]
aaa żeby dodać wątek z Home Assistant, tak?

**Reply by piotrux** [reply]
Tak, telefon i słownik = tragedia

**Comment by piotrux** [comment]
Odpaliłem virtual thermostat I ładnie czyta temperatury zewnętrzną i wewnętrzną, teraz pytanie jak sterować temperaturą miarkownika a nie on/ off pieca?

**Reply by krzychoooo** [reply]
Tu na githabie, na pierwszej stronie w read.me jest spis filmów i już jest film jak z domoticza sterować miarkownikiem.

**Reply by piotrux** [reply]
Dziękuję,

**Comment by piotrux** [comment]
Witam,
Mam prośbę do Pana Krzysztofa bo słabo mi idzie nauka domoticza, czy jest możliwość aby mi pan trochę podpowiedział.
Wymyśliłem że najlepiej będzie zrobić sterowanie na podstawie krzywej grzewczej, tj dodam sobie wirtualny termostat z "temp wew zadana" w pokoju i z wzoru ("współczynniki krzywej grzewczej"*("temp wew zadana"-("temp zew z netatmo")))+"temp wew zadana" wyliczę temp pieca ale nie radzę sobie z napisaniem kodu.

**Reply by krzychoooo** [reply]
Zdrowia w nowym roku.
Jeśli zaczynać programowanie w Domoticz to moim zdaniem od razu wejść w nowe rozwiązania czyli DzVents. 
Tu są podstawy [DzVents](https://www.domoticz.com/wiki/DzVents:_next_generation_Lua_scripting#Thermostat_set_point)
Trochę tego dużo i na początku trochę odpychające ale warto.
Przykład skryptu który przed chwilą sprawdziłem i działa:
```
return {
    logging = {
		-- level = domoticz.LOG_DEBUG, -- Select one of LOG_INFO, LOG_DEBUG, LOG_ERROR, LOG_FORCE to override system log level
		level = domoticz.LOG_DEBUG,
		-- marker = "(Setpoint)"
			  },
	on = {
		devices = {
			'Termometr pokój TV'
		}
	},
	execute = function(domoticz, device)
		domoticz.devices('Ustawiona temperatura pieca').updateSetPoint(120 - 3*domoticz.devices('Termometr pokój TV').temperature)
		domoticz.log('Ustawiona na kotle '..domoticz.devices('Ustawiona temperatura pieca').setPoint, domoticz.LOG_INFO)
	end
}
```
Ustawia temp pieca wg. wzoru Tpiec = 120-3*Temperatura_pokoju_TV
Skrypt wyzwalany jest przy każdej zmianie urządzenia 'Termometr pokój TV'
Może to naprowadzi jak to zrobić

**Reply by piotrux** [reply]
hmmm u mnie nie działa:
return {
    logging = {
		-- level = domoticz.LOG_DEBUG, -- Select one of LOG_INFO, LOG_DEBUG, LOG_ERROR, LOG_FORCE to override system log level
		level = domoticz.LOG_INFO,
		-- marker = "(Setpoint)"
			  },
	on = {
		devices = {
			'Outdoor_Temperature'
		}
	},
	execute = function(domoticz, device)
		local setpoint = (120 - 3*domoticz.devices('Outdoor_Temperature').temperature)
        local url= 'http://192.168.0.17/control?cmd=event,setTemp='..(setpoint.state)..''
        print(url)
        domoticz.openURL(url)
		domoticz.log('Ustawiona na kotle '..domoticz.devices('Termostat').setPoint, domoticz.LOG_INFO)
	end
}

**Reply by krzychoooo** [reply]
setpoint to liczba a nie obiekt więc nie można tak:
`setpoint.state`
liczba nie zawiera już w sobie jakiś elementów do których można się odwołać kropka
lepiej użyć 
`tostring(setpoint)`

**Reply by piotrux** [reply]
Dziękuje za wszystko działa :) zrobiłem na krzywej grzewczej

**Comment by zbyszkola** [comment]
Mam na smartfonie wgraną apkę Domoticz a le utknąłem na samym początku,nie wiem jakiadres  serwera,nazwe użytkownika,hasło wpisć?
Apka Miarkownik01 działa na IP 192.168.4.1 i jest ok.W przeglądarkę wpisuje to ip 192.168.4.1/setup to wchodzę w ustawienia Miarkownik.Na laptopie na windows7 też mam wgrany program Domoticz (2020.2) gdzie też nie wiem co na początku powpisywać.Jako takiego poradnika początkującego nie znalazłem.

**Reply by piotrux** [reply]
Na androida jest tylko client domoticza. Jak masz na Windows to wpisz w przeglądarce http://localhost:8080. W apce android podaj IP komputera z Windows

**Reply by zbyszkola** [reply]
skupie się narazie na Windowsie7,po wpisaniu tego dalej nie wiem co zrobić,gdzie co wpisać...masakra z tym Domoticzem...

**Reply by piotrux** [reply]
W przeglądarce internetowej. Jak nie zadziała spróbuj http://127.0.0.1:8080. Jak nie zadziała, sprawdź czy masz ikonkę domoticza przy zegarku na pasku.

**Reply by zbyszkola** [reply]
tak mam ikonkę koło zegara,Domoticz mi się uruchomił,nawet utworzyłem na początku w Dummy ale usunąłem i dałem na zdalny serwer wczesniej na wewnętrzny ale tez usunąłem,wsumie to nie wiem jak tam ustawić.

**Reply by piotrux** [reply]
Jak działa domoticz to film od Pana Krzysztofa tłumaczy jak podpiąć (dodać) miarkownik. https://youtu.be/bdN2KOH_0wY

**Reply by krzychoooo** [reply]
> Apka Miarkownik01 działa na IP 192.168.4.1

Czy na pewno poprawnie jest skonfigurowany miarkownik? Czy miarkownik i apka na telefonie działają poprzez router?

**Reply by krzychoooo** [reply]
Podczas normalnej pracy używanie dostępu do miarkownika poprzez sieć miarkownik to zły pomysł. To nie będzie działać poprawnie.
Miarkownik gdy nie może się zalogować do sieci routera to tworzy sieć "Miarkownik" ale tylko na jakiś czas, następnie znowu próbuje się zalogować do sieci routera i tak wkoło.
Wczoraj wrzuciłem kolejny film o konfiguracji Miarkownika https://www.youtube.com/watch?v=oIrx52zjV-o.
Jak dalej problemy proszę zadzwonić tel z aukcji. Chętnie pomogę.

**Reply by krzychoooo** [reply]
> .Na laptopie na windows7 też mam wgrany program Domoticz (2020.2)

W systemie/sieci/domu powinien być zainstalowany domoticz tylko w jednym miejscu/urządzeniu. 
No chyba że dużo eksperymentujemy ale na początek polecam **jeden**   Domoticz na sieć.

**Reply by zbyszkola** [reply]
krzychoooo i piotrux wielkie dzięki za pomoc,odezwę się jeszcze,albowiem teraz pracuję na popołudniówki i z rana tak trochę mało czasu.

**Comment by zbyszkola** [comment]
Witam,udało mi się w Domoticzu na lapku z linuxem utworzyć czujkę temperatury lecz aktualnej temperatury nie pokazuje.Co prawda teraz w pracy jestem i nie mam dostepu do lapka ale pytanie:
-czy w apce Domoticz lub innej na smartfonie  jest możliwość podglądu temperatury z miarkownika na piecu c.o.?
- czy pod domoticz można ustawić i mieć możliwość włączania i wyłączania gniazda wifi (w apce SmartLife tak mam-4 gniazda)?
-ewentualnie pod jakąś apkę podłączyć miarkownik żeby móc zdalnie obserwować temperaturę pieca?

**Reply by krzychoooo** [reply]
W domoticz każde urządzenie posiada swój IDX. Ten IDX należy wprowadzić w miarkowniku. 

1. w przeglądarce otwieramy adres IP miarkownika
2. w zakładce Controllers dodajemy Domoticz HTTP podając adres IP domoticza "Controller IP"
3. W zakładce Devices wchodzimy w edycję "Environment - DS18b20" 
4. Zaznaczamy "Send to Controller" oraz IDX takie jak w domoticzu ma kontrolka temperatury

**Reply by zbyszkola** [reply]
Albo ja jestem oporny albo ten Domoticz,nie pokazuje dalej mi temperatury.Robię według tego opisu wyżej i dalej nic.Utworzyłem już sobie Miarkownik i Termometr_1 o IDX 3 i 4 w Domoticz/Konfiguracja/Sprzęt natomiast w Domoticz/Konfiguracja/Urządzenia  IDX to 2 i 3  więc w Miarkowniku/Devices/Edit-Termometr_1/IDX już daje po kolei 2,3,4 a nawet w Miarkownik/Tools/Reboot i dalej brak wskazań temperatury.Na apce Miarkownik01 wskazania temp.są.Co takiego źle robię.Domoticza ma zamontowanego na Ubuntu Mate 20.10

**Reply by krzychoooo** [reply]
Czy na pewno w miarkowniku jest podany poprawny IP Domoticza?
Pokażę na przykładzie
Mam w domoticz->konfiguracja->sprzęt
![Screenshot_2021-03-12 Dom](https://user-images.githubusercontent.com/7173366/110924740-b09ae500-8322-11eb-9c4e-d911362af217.png)
Tworzę wirtualną czujkę (dummy device) i pojawia mi się:
W Domoticz->konfiguracja->urządzenia
![Screenshot_2021-03-12 Dom2](https://user-images.githubusercontent.com/7173366/110924842-ce684a00-8322-11eb-95d6-9682218d12c8.png)
Temperatura pieca z IDX=1
mój domoticz ma IP 192.168.8.111
przechodzę do miarkownik->tools i w lini komend wpisuję:
![Przechwytywanie](https://user-images.githubusercontent.com/7173366/110925341-5c443500-8323-11eb-8b34-9a6efef2102a.PNG)
Daję submit
i jeśli wszystko jest z adresami oK to w domoticzu pojawia mi się temperatura pieca o wartości 123

**Reply by krzychoooo** [reply]
W miarkownik->tools->Log
mamy podgląd co się dzieje(najlepiej **Wyłączyć** apkę na telefonie). Jeśli w którejś linijce jest error to skopiuj ją i wklej tutaj.
Postaram się pomóc

**Reply by krzychoooo** [reply]
Samego domoticza można przetestować wpisując w adresie przeglądarki:
"http://192.168.8.111:8080/json.htm?type=command&param=udevice&idx=1&nvalue=0&svalue=123" 
oczywiście zmieniając na własny IP Domoticza oraz poprawny IDX

**Reply by zbyszkola** [reply]
Dobra,już się pojawiła temperatura123-zrobiłem tak jak pisało-idx to 2
Domoticz przetestowałem i jest ok: "status" : "OK", "title" : "Update Device"
Skąd brać ten wpis w tą linię komend-bo jego tylko nie miałem i dlatego był problem.
No i co dalej z tą temp 123,chodzi mi o rzeczywistą temperaturę.
Co do błędów w logach Miarkownika to mam tylko to(na żółto mi wyświetla)a errorów nie ma:
Conn. IP Init
364703: HTTP : C008 connection failed (100/0?)
365963: HTTP : C008 connection failed (104/0)
366210: HTTP : C008 connection failed (105/0)
366440: HTTP : C008 connection failed (106/0)
366671: HTTP : C008 connection failed (107/0)
366900: HTTP : C008 connection failed (108/0)
367148: HTTP : C008 connection failed (109/0)
367377: HTTP : C008 connection failed (110/0)
775553: HTTP : C008 connection failed (aż do 220/0)

**Reply by krzychoooo** [reply]
Jaki adres IP ma miarkownik?
Jeśli 192.168.4.1 to miarkownik nie przeszedł procesu instalacji do routera. Wtedy Domoticz jest w jednej sieci tworzonej poprzez router a miarkownik w drugiej sieci tworzonej przez miarkownik. Nie ma możliwości wymiany danych pomiędzy dwoma różnymi sieciami.
Miarkownik i Domoticz MUSZĄ być w tej samej sieci, tworzonej przez router.
Można dzwonić do mnie tel z aukcji.

**Comment by zbyszkola** [comment]
Znowu jeatem w pracy i nie mam dostępu do Domoticza i Miarkownika,ale jest
tak:
Miarkownik 192.168.1.2
Domoticz 192.168.1.10





pt., 12 mar 2021, 16:18 użytkownik krzychoooo ***@***.***>
napisał:

**Reply by krzychoooo** [reply]
Adresy wyglądają OK.
Wklej tutaj ekrany z takich adresów:
192.168.1.2/controllers?index=1
192.168.1.2/devices?index=1&page=1

**Comment by zbyszkola** [comment]


**Reply by krzychoooo** [reply]
Nic nie wyświetla mi się

**Reply by krzychoooo** [reply]
Fotki tutaj można wklejać w tym polu:
![Przechwytywanie3](https://user-images.githubusercontent.com/7173366/111028546-941cac80-83f7-11eb-9556-be78811389a6.PNG)

**Reply by zbyszkola** [reply]
Dwa screny:

![scr1](https://user-images.githubusercontent.com/79381718/111029884-70f5fb00-83ff-11eb-8817-d5a8cdee542f.jpg)
![scr2](https://user-images.githubusercontent.com/79381718/111029886-72272800-83ff-11eb-8039-c16324e7a1ca.jpg)

**Reply by krzychoooo** [reply]
Zmień kontroler z Generic HTTP na Domoticz HTTP

**Reply by zbyszkola** [reply]
zmieniłem w Miarkowniku/Controllers i teraz mam 2 takie same...
![dom1](https://user-images.githubusercontent.com/79381718/111031515-3ba1db00-8408-11eb-98f1-91cb41275435.jpg)

**Reply by zbyszkola** [reply]
zrobiłem że teraz mam jeden Domoticz HTTP i dalej temperatura 123
![dom2](https://user-images.githubusercontent.com/79381718/111031712-2b3e3000-8409-11eb-967c-cd9051d9c8bd.jpg)
Byłem teraz w LOGu Miarkownika i żadnych błędów-Error nie ma

**Reply by krzychoooo** [reply]
najlepiej jak by był na pozycji 1
chyba że w edycji termometru zmienisz na kontroler nr3

**Reply by krzychoooo** [reply]
![Przechwytywanie4](https://user-images.githubusercontent.com/7173366/111031898-1910c180-840a-11eb-9aa3-f65340c44557.PNG)

**Reply by krzychoooo** [reply]
W tools to pole dla komend normalnie ma być puste. To jest tylko do testowania sobie komend których używa się w rules

**Reply by zbyszkola** [reply]
nr kontrolera to 1 a idx 2...
![dom3a](https://user-images.githubusercontent.com/79381718/111032403-5c6c2f80-840c-11eb-9297-354b76e7d349.jpg)
![dom3b](https://user-images.githubusercontent.com/79381718/111032404-5d9d5c80-840c-11eb-8d2e-075536ba6ede.jpg)
![dom4](https://user-images.githubusercontent.com/79381718/111032405-5e35f300-840c-11eb-8a0f-09c1060bd324.jpg)
![dom5](https://user-images.githubusercontent.com/79381718/111032406-5e35f300-840c-11eb-8ef9-b7ba77165231.jpg)
![dom6](https://user-images.githubusercontent.com/79381718/111032407-5ece8980-840c-11eb-867a-dcef604de4b8.jpg)
![dom7](https://user-images.githubusercontent.com/79381718/111032408-5f672000-840c-11eb-98ec-27e063443dd2.jpg)
![dom8](https://user-images.githubusercontent.com/79381718/111032829-76a70d00-840e-11eb-9de0-f138620f228d.jpg)

**Reply by krzychoooo** [reply]
Czyli to pole odpowiedzialne za kontroler nr1 jest puste teraz?
![Przechwytywanie5](https://user-images.githubusercontent.com/7173366/111033126-fa152e00-840f-11eb-84e7-7bfb94cc1ccf.PNG)

**Reply by zbyszkola** [reply]
![domoti1](https://user-images.githubusercontent.com/79381718/111034157-8e818f80-8414-11eb-9e23-465689d6b161.jpg)
Zrobiłem od nowa,tamto pousuwałem i mam od początku Domoticz HTTP na 1 pozycji ,jest z IDX 2, Enabled w Controllers zachaczone (włączone)
![domoti1](https://user-images.githubusercontent.com/79381718/111034157-8e818f80-8414-11eb-9e23-465689d6b161.jpg)

**Reply by krzychoooo** [reply]
Czy komenda wysłana z tools przesyła temperaturę?
SendToHTTP 192.168.1.10,8080,/json.htm?param=udevice&type=command&idx=2&nvalue=0&svalue=[Termometr_1#Temperature]

**Reply by krzychoooo** [reply]
Wydaje się być teraz wszystko OK
W Tools->Settings->Save
pobiera plik konfiguracji i prześlij mi go na fajnysoft@gmail.com  może jeszcze coś zauważę.

**Comment by zbyszkola** [comment]
Nie,temperatura nie pojawia się.pomału mam dość tego Domoticza.Ale problem dalej nastąpił jak się przelogowałem na windows7 i z ustawieniem temperatury miarkownika,zajżałem też do logów programu Domoticz i w ogole nie ma zakładki Błędy.Przelogowałem się z powrotem na Linux ubuntu mate i tam jest ta zakładka,poniżej screny z tego i inne.

![demoti2](https://user-images.githubusercontent.com/79381718/111036759-28e7d000-8421-11eb-9d93-2f6468eee783.jpg)
![demoti3](https://user-images.githubusercontent.com/79381718/111036761-2a18fd00-8421-11eb-8331-5421a0b8736d.jpg)
![demoti4](https://user-images.githubusercontent.com/79381718/111036762-2ab19380-8421-11eb-9923-02428fe946e8.jpg)
![domoti5logi](https://user-images.githubusercontent.com/79381718/111036763-2b4a2a00-8421-11eb-93dc-6fa5e8002bfe.jpg)

to Save zaraz poszukam i prześlę....

**Reply by krzychoooo** [reply]
Te błędy domoticz sypie bo coś jest źle w skrypcie "Script #1".   Domoticz->konfiguracja->więcej opcji->zdarzenia

**Reply by krzychoooo** [reply]
A to ostrzeżenie na czerwono w Controllers mówi że przekroczono dopuszczalną liczbę zapisów na dzień (ochrona pamięci flash). 
Trzeba zresetować miarkownik przez zasilanie.

**Comment by zbyszkola** [comment]
Zresetowałem Miarkownik jak pisałeś-i coś nowego się dowiedzieliśmy o dużej liczbie zapisów.
a tu ze Script #1
return {
	on = {
		devices = {
			'myDevice'
		}
	},
	execute = function(domoticz, device)
		domoticz.log('Device ' .. device.name .. ' was changed', domoticz.LOG_INFO)
	end
}
<xml xmlns="http://www.w3.org/1999/xhtml"><block type="temperaturevariables" id="7gS^3sw[9:Nkio]4:2m^" x="-308" y="10"><field name="Temperature">2</field></block><block type="security_status" id=":@dd9~/FU*tM1Ey$4{VY" x="-362" y="19"><field name="OP">EQ</field><field name="Status">0</field></block><block type="temperaturevariables" id="[]{vyEe?jN{i+sQIY,+E" x="-323" y="19"><field name="Temperature">2</field></block><block type="temperaturevariables" id="Gg(+o)urW5C}zXh?E6IY" x="-90" y="20"><field name="Temperature">2</field></block></xml>

Daje ponizej  zrzut tego Scripta bo chyba źle skopiowało:
![script](https://user-images.githubusercontent.com/79381718/111037810-1d4ad800-8426-11eb-90e6-6a71ed9db580.jpg)

**Reply by krzychoooo** [reply]
Załadowałem ten config przesłany na email i zmieniłem jedynie IP Domoticza i adres termometru(każdy termometr ma swój) i ruszyło od razu.
Czy na pewno komputer na którym jest Domoticz i Miarkownik są w tej samej sieci? Miarkownik->Config->SSID

**Comment by zbyszkola** [comment]
Internet mam z Netii i wszytko działa po wifi w jednej sieci,jak i tv,sat,laptop,miarkownik,Domoticz na lapku...
Pogrzebałem jeszcze w tym Domoticzu zajrzałem tam w logi i dalej błąd ze Script 1

**Reply by krzychoooo** [reply]
Nie wiem co to za kod  między <xml    a   </xml> 
J bym to skasował.

**Reply by krzychoooo** [reply]
W Domoticz->Konfiguracja->Ustawienia jak masz to ustawione:
![Przechwytywanie6](https://user-images.githubusercontent.com/7173366/111040057-a9163180-8431-11eb-8451-e3303de6c3b7.PNG)

**Comment by zbyszkola** [comment]
tamto skasowałe,w tej sieci lokalnej jest taki adres 192.168.1.0  i taki miałem z automatu-nie wpisywałem tego

**Reply by krzychoooo** [reply]
192.168.1.0 to jest najczęściej adres routera
powinieneś mieć takie ustawienie
127.0.0.*;192.168.1.\*

**Reply by krzychoooo** [reply]
127.0.0.*;192.168.1.\*

**Reply by krzychoooo** [reply]
pogooglaj na temat "Domoticz ustawienia poinstalacyjne"

**Comment by zbyszkola** [comment]
zmieniłęm to IP tam

 dalej błąd wywala w Script1

 2021-03-13 19:21:00.423 Error: dzVents: Error: (3.0.2) error loading module 'Script #1' from file '/home/ubuntu/domoticz/scripts/dzVents/generated_scripts/Script #1.lua':
2021-03-13 19:21:00.423 ...domoticz/scripts/dzVents/generated_scripts/Script #1.lua:11: <eof> expected near 'xmlns'
2021-03-13 19:22:00.436 Error: dzVents: Error: (3.0.2) error loading module 'Script #1' from file '/home/ubuntu/domoticz/scripts/dzVents/generated_scripts/Script #1.lua':
2021-03-13 19:22:00.436 ...domoticz/scripts/dzVents/generated_scripts/Script #1.lua:11: <eof> expected near 'xmlns'
2021-03-13 19:23:00.446 Error: dzVents: Error: (3.0.2) error loading module 'Script #1' from file '/home/ubuntu/domoticz/scripts/dzVents/generated_scripts/Script #1.lua':
2021-03-13 19:23:00.446 ...domoticz/scripts/dzVents/generated_scripts/Script #1.lua:11: <eof> expected near 'xmlns'

**Comment by krzychoooo** [comment]
po zmianie na 127.0.0.\*;192.168.1.\* odczytuje temperaturę?

**Reply by zbyszkola** [reply]
nie ,dalej pokazuje 123 st.C

**Reply by krzychoooo** [reply]
Cholernie odporny ten domoticz :)
Skoro ten adres: http://192.168.1.10:8080/json.htm?type=command&param=udevice&idx=2&nvalue=0&svalue=123 wprowadzony z przeglądarki działa to pytanie
czy ta przeglądarka odpalona na tym samym kompie co zainstalowany domoticz? Jeśli tak to znaczy że lokalnie działa
można by spróbować z innego kompa albo telefonu oczywiście zmieniając końcówkę 123 na inną wtedy sprawdzimy czy działa przez sieć routera

**Reply by zbyszkola** [reply]
Na lokalnej sieci ten adres działa na innej (wifi ze smartfona) nie działa.Domoticz i Miarkownik też nie działają wtedy.
Martwią mnie te błedy w tym Script #1 w logach,może swój mi podeślij?jak jest inny...

>>>>>>edit: ja mam koncówkę routera inną niż wcześniej pisałes (254)........


![scre2](https://user-images.githubusercontent.com/79381718/111042803-6eb39100-843f-11eb-9acb-f90d9ea25629.jpg)
![scren przegladarki](https://user-images.githubusercontent.com/79381718/111042804-6fe4be00-843f-11eb-995a-26972c3605a6.jpg)

**Comment by zbyszkola** [comment]
Walcze dalej z tym Domoticzem,na windowsie7 skasowałem go,przeczyściłem cccleanerem rejestr,uruchomiłem od nowa lapka,wgrałem nowego Domoticza wers. na windows.
I tak Domoticz uruchamia się na innym adresie niż wczoraj a mianowicie na http://127.0.0.1:8080 ustawiłem w nim Miarkownik o IDX1 ale temperatury dalej nie pokazuje, w Domoticz/Logi-błędów brak w Domoticz/wiecej Opcji/Zdarzenia brak jest Script1 a wczoraj był ten Script...

sprawdziłe poprzednie IP http://192.168.1.10:8080 i też na tym Domoticz działa.

**Reply by krzychoooo** [reply]
Adres 127.0.0.1 to adres lokalnej maszyny w komputerze(local host) i obowiązuje tylko wewnątrz komputera.
Domoticz w sieci działa pod takim ip jaki został przydzielony komputerowi przez router. [https://pl.wikipedia.org/wiki/Localhost]

**Reply by krzychoooo** [reply]
W Domoticz->Konfiguracja->Ustawienia jak masz to ustawione?:
![Przechwytywanie6](https://user-images.githubusercontent.com/7173366/111040057-a9163180-8431-11eb-8451-e3303de6c3b7.PNG)
To jest bardzo ważne ustawienie.

**Comment by zbyszkola** [comment]
YES....udało się...
Dzisiaj tak z doskoku w wolnej chwili grzebałem w Domoticzu,różne wpisy robiłem o to co pytasz wkońcu zadziałało na 127.0.0.*;192.168.1.* ale to przez to że walczyłem w Dmoticzu ze Sprzętem i Wirtualnymi Czujnikami  oraz z przestawieniem IDX w Miarkowniku.Teraz mam 3 Urządzenia w Konfiguracji Domoticza-1 napewno muszę usunąć chyba że dwa.W każdym bądź razie działa.Poniżej parę screnów:
![do2](https://user-images.githubusercontent.com/79381718/111072694-fe0e8200-84db-11eb-947b-9446d49773c0.jpg)
![do3](https://user-images.githubusercontent.com/79381718/111072697-ff3faf00-84db-11eb-95ae-c3e0d489fe03.jpg)
![do4](https://user-images.githubusercontent.com/79381718/111072698-ffd84580-84db-11eb-8d4f-789a1289f8b3.jpg)
![do5](https://user-images.githubusercontent.com/79381718/111072699-ffd84580-84db-11eb-86e4-4eaea6633df1.jpg)
![do6](https://user-images.githubusercontent.com/79381718/111072700-0070dc00-84dc-11eb-9f83-addf60026b21.jpg)

**Reply by zbyszkola** [reply]
Ustawiłem już sobie w tym Domoticzu,napewno najważniesze jest to IP o którym pisałeś ale takze ustawienia czujki-w tym się gubię i dlatego jest taki problem.W logach błedów brak,w Zdarzeniach nie ma Scriptu1 i zadnego innego a Domoticz na windows 7 działa.
![do8](https://user-images.githubusercontent.com/79381718/111073114-caccf280-84dd-11eb-83eb-38dbe31e35da.jpg)
![do9](https://user-images.githubusercontent.com/79381718/111073115-cbfe1f80-84dd-11eb-8396-1d121bb454d0.jpg)
![do10](https://user-images.githubusercontent.com/79381718/111073118-cbfe1f80-84dd-11eb-9f51-dbc5676e1a73.jpg)

**Reply by krzychoooo** [reply]
Super. Gratulacje za wytrwałość :)

**Reply by krzychoooo** [reply]
Te skrypty są nie potrzebne do działania domoticza. One są po to aby wprowadzić własną logikę np:
Jest czujka temperatury w pokoju i na jej podstawie, ustawiana jest temperatura w miarkowniku.

**Reply by zbyszkola** [reply]
Gratulację i Podziękowania to się Tobie **krzychoooo** należą i także za wytrwałość-do takich jak ja...
Normalnie tobie się należy nie piwo ale flacha-ale to może kiedyś :>)  ......
Czujki temp.w pokoju nie mam-to się obędzie bez tego.Zamontowałem czujkę drukarki HP wifi -chodzi mi tylko on/off bo jest w drugim pokoju i jak zwykle ustawienia wirtualne czujki-ale narazie to w wolnej chwili w ramach zabawy-zapoznania się.
Mam jak wspominałem wczesniej gniazdka po wifi i także kamerkę monitoringu po wifi-jest możliwość podłaczenia tego w Domoticzu?Chodzi mi tylko o on/off - włączania wyłączania 2 mat grzewczych na podczerwień. Oczywiście to nie teraz tylko później mam zamiar taki projekt zrealizować.

edit:
wszystko fajnie w tym Domoticzu,wiem jak dodać wirtualne czujki ale co z dodaniem IP danego urządzenia?Muszę na każdym IP urządzenia mieć ten "programik" (czy jak tam się zwie)jak Miarkownik?No bo nie wiem gdzie w jaką pozycję wklepać IP?

**Reply by krzychoooo** [reply]
Urządzenia wysyłają dane pod określony IDX i to jest dla domoticza istotne.
Te gniazdka WiFi to jaki producent? symbol?

**Reply by zbyszkola** [reply]
te gniazdka to Vanseng ( https://allegro.pl/oferta/inteligentne-gniazdo-wifi-vanseng-wtyczka-alexa-1-10028660315 ) IP odczytałem z apki SmartLife,szkoda że na tej apce Miarkownika nie mogę podpiąć.

Poszła aktualizacja Domoticza na linuxie mam V2020.2 i jest problem z odpaleniem Domoticza-akurat wyczytałem na facebooku o tym.I chyba ciekawa rzecz którą tam wyczytałem-skopiowałem pewną wypowiedź,daje poniżej:

'' ja wczoraj zaktualizowałem jak napisałem poniżej i domotycz był off line w efekcie kolega siedział pół nocy i stwierdził że zainstaluje mi wersję 2020.1 działa bez zażutów dodatkowo w nowej wersji można bez stałego IP łączyć się z domotyczem poza swoją siecią wifi niema tego złego co by na dobre nie wyszło''

Domoticz po aktualizacji do V2020.2 na Ubuntu Mate 20.10 nie działa a na Windows7 działa.

**Reply by krzychoooo** [reply]
Wydaje mi się że to kopia SonOff S20 a jak wgrać ESPEASY do niego jest tutaj 
[http://blog.chodorow.ski/2018/01/sonoff-s20-esp-easy.html](https://www.youtube.com/watch?v=GbaMF6zfQZU)
i tutaj
[https://www.youtube.com/watch?v=GbaMF6zfQZU](https://www.youtube.com/watch?v=GbaMF6zfQZU)

**Reply by zbyszkola** [reply]
to chyba lipa z tymi gniazdkami w Domoticzu,czytałem ten blog-ja mam raczej nowego typu a oprogramowanie działa tylko na tych starego typu(2018r). Czyli każde urządzonko z wifi powinno mieć wgrane oprogramowanie ESP Easy...żeby Domoticz widział jego IDX... to tak pokrótko rozumując.To dla drukarki HP z wifi też by trza to ESP Easy wgrać?

tu dla tych co może ich zainteresować: ESP Easy - konfiguracja dla Domoticz http://blog.chodorow.ski/2018/01/esp-easy-konfiguracja-dla-domoticz.html#comment-form

**Reply by krzychoooo** [reply]
> Czyli każde urządzonko z wifi powinno mieć wgrane oprogramowanie ESP Easy...żeby Domoticz widział jego IDX... to tak pokrótko rozumując.

Nie.  Ale musi być zgodne z protokółami Domoticza. Domoticz również odbiera w MQTT więc wszystko co posiada MQTT będzie działać.
Niestety na rynku mamy mnóstwo urządzeń które współpracują tylko ze swoim softem, i takie coś musi mieć zmieniony firmware aby działało z Domoticz.
